packages/taquito/src/operations/types.ts

```diff
@@ -563,6 +566,53 @@ export interface FailingNoopParams {
   basedOnBlock: BlockIdentifier;
 }
 
+export interface InternalStakingParams {
```

Comment:
what's the reason of creating ` InternalStakingParams` with source required?

Reply:
It is used in `createStakingOperation` to make sure the calling code has added source to the `params` provided by the Dapp developer